nucliadb/src/nucliadb/reader/api/v1/download.py

```diff
@@ -342,14 +345,16 @@ async def download_api(sf: StorageField, headers: Headers, inline: bool = False)
                 headers={"Content-Range": f"bytes */{file_size}"},
                 status_code=416,
             )
-        status_code = 206
         logger.debug(f"Range request: {range_request}")
         extra_headers["Content-Length"] = f"{range_size}"
         extra_headers["Content-Range"] = f"bytes {start}-{end}/{file_size}"
-        download_headers["Range"] = range_request
+        range_start = start
+        range_end = end
 
     return StreamingResponse(
-        sf.storage.download(sf.bucket, sf.key, headers=download_headers),  # type: ignore
+        sf.storage.download(
+            sf.bucket, sf.key, range_start=range_start, range_end=range_end
```

Comment:
I'd be tempted to write a small `Range` class that wraps `range_start` and `range_end` since it's passed along so much. It can also have a few helper functions since some operations are repeated here and there (converting to string, checking if any of start/end is None, etc.).

On the other hand, it might be a big change for not too much gain, and being explicit is also nice. So just leavign this as a suggestion, up to you!

Reply:
I like it!